### ISSUE TITLE
Generating MARC records from C0140 EAD component records/archival objects in Resource records 

### ISSUE THREAD

**Issue by faithc** [pulibrary/aspace_helpers#181]
Request for MARC records to be created for all individual component records/archival objects in C0140 (General Manuscripts Miscellaneous Collection) rather than generating a collection-level record, to allow for increased discoverability of materials that represent standalone items.

This relates to pulibrary/pulfalight#540.

**Comment by regineheberlein** [comment]
@faithc 

> This generation currently happens now with EADs in SVN 

Can you point me to the file(s) in SVN that used to generate MARC item records from EAD? Is it some kind of XSLT? Was it run manually?

**Comment by regineheberlein** [comment]
@faithc The quote is from a comment you left on https://github.com/pulibrary/pulfalight/issues/540, which you referenced above, so it seems like you're saying there is an existing SVN workflow that we should emulate. Could you review your comment in case it sparks any memory? I know this was a long time ago.

**Comment by regineheberlein** [comment]
@faithc 
Also, can you provide a mapping or sample MARC record to show what fields you want included / what you want the final output to look like?

**Comment by regineheberlein** [comment]
@faithc 
I just need a little more information please. Most component records don't have creators, just titles, so who should the creator be? Which fields specifically do you want included? I don't think the collection-level routine carries over; we went through quite a lot of back and forth to map and filter the fields. Again, an example or field-by-field mapping would be helpful.

**Comment by faithc** [comment]
Ok, will do.

**Comment by faithc** [comment]
Might you be able to share the documentation/mapping that was created for collection-level MARC records? Might be easiest for us to work from that as a baseline.

**Comment by regineheberlein** [comment]
Sure! If you want to know more about the collection-level process, you may also want to check in with Lynn.

Umbrella issue: https://github.com/pulibrary/aspace_helpers/issues/2

Code: https://github.com/pulibrary/aspace_helpers/blob/main/reports/aspace2alma/get_MARCxml.rb

**Comment by faithc** [comment]
Thanks! And will do.

**Comment by faithc** [comment]
Hey Regine- I checked in with Lynn. She said that as far as she's aware, IT folks have been using the standard MARC XML export with its built-in mapping. Is that sufficient for this use case as well? Perhaps if there are differences between the two record types/workflows, we can hash those out? For example, I would think that similar to collection-level records, if there's no creator associated with an archival object, then that MARC field doesn't get populated.

**Comment by regineheberlein** [comment]
Hi Faith. "IT folks" in this case is myself, so I can say with certainty that no, we didn't just import the ASpace export into Alma. Rather, we exported MARC-XML and then quite significantly tweaked the fields based on Lynn's (and other folks' she was working with) requests.

For this project, none of that applies, as MARC-XML exports in ASpace are only available on the collection level. For the component level, we'll be working off of the JSON export.

**Comment by faithc** [comment]
Ok, sorry. I wasn't sure who all was involved in that project.

**Comment by regineheberlein** [comment]
> Physical Location (can this be pulled from the collection-level note?) | 982
> -- | --

Can you provide a little more information about this? 

The collection is split between two locations, both of which are reflected on the collection level--so pulling the location from the collection level for the item-level bib records doesn't make sense to me. We can pull it from the linked container record for each object.

Do you want the location in a bib 982? Do you want a holdings record created? Again, a MARC example would be helpful.

**Comment by regineheberlein** [comment]
> . | .
> -- | -- 
> Agent/Creator/Persname or Famname | 100
> Agent/Creator/Corpname | 110
> Agent/Subject | 600 
> Agent/Creator | 700 

Can you elaborate on this please? 
* Do you want a 6xx + 7xx for any creator? Or do you only want the 600 / 700 (i.e. personal names)? 
* Do you want the 6xx / 7xx when the agent is linked with the corresponding role in ASpace, or are you asking that we construct the 6xx / 7xx?

**Comment by faithc** [comment]
> > Physical Location (can this be pulled from the collection-level note?)
> > 982
> 
> Can you provide a little more information about this?
> 
> The collection is split between two locations, both of which are reflected on the collection level--so pulling the location from the collection level for the item-level bib records doesn't make sense to me. We can pull it from the linked container record for each object.
> 
> Do you want the location in a bib 982? Do you want a holdings record created? Again, a MARC example would be helpful.

Pulling the info from the linked container record makes sense to me. (I think there's a ticket to have this happen at the collection level as well.) 

Do we need physical location to be added to bib 982 if that info is in the holdings record? The only reason we thought a bib 982 was needed is because this is in included in collection-level MARC records. On the flip slide, should we have this information only in bib 982 and not create a holdings record since we don't want users to be able to request items from the catalog? Is a holdings record required?

**Comment by faithc** [comment]
> > .
> > .
> > 
> > 
> > 
> > 
> > Agent/Creator/Persname or Famname
> > 100
> > 
> > 
> > Agent/Creator/Corpname
> > 110
> > 
> > 
> > Agent/Subject
> > 600
> > 
> > 
> > Agent/Creator
> > 700
> 
> Can you elaborate on this please?
> 
> * Do you want a 6xx + 7xx for any creator? Or do you only want the 600 / 700 (i.e. personal names)?
> * Do you want the 6xx / 7xx when the agent is linked with the corresponding role in ASpace, or are you asking that we construct the 6xx / 7xx?

Yes, sorry, we would like a 6xx + 7xx for any creator; and yes, the a 6xx + 7xx when the agent is linked with the corresponding role. Is that right, @LynnDurgin?

**Comment by faithc** [comment]
I tried to create a sample MARC record from the mapping we did: https://docs.google.com/document/d/1iSgOT3CMJBo7_BacrJz7udJOywNAPNlDTihssWNZ1Pw/edit. Is this helpful/what you're looking for @regineheberlein?

**Comment by LynnDurgin** [comment]
@faithc Yes, that is my understanding.

**Comment by regineheberlein** [comment]
Going back to the 7xx, can you provide the role values in ASpace that should trigger the creation of the 7xx please?

**Comment by LynnDurgin** [comment]
I don't think we are looking to the relator values in this case, but I would like @faithc to confirm.

**Comment by LynnDurgin** [comment]
Regarding whether to have holdings in these records, I would favor having them-- they display the location in a highly visible way, and our collection level records have them, so there would be a level of consistency. There shouldn't be any danger of people requesting these materials through the catalog because, as I understand it, there will be no item records associated with the holdings and the request button will be suppressed.

**Comment by doncat** [comment]
(After lurking a long time)
Regarding Lynn's comment about holdings: I believe there has to be "inventory" of some kind for a record to display in Blacklight.  In this case the holdings will be kind of fake, but I think it still has to be there.

**Comment by regineheberlein** [comment]
Yeah, Alma creates a dummy.

Regine Heberlein (she/her)
Library IT Data Analyst
609-258-6156
***@***.******@***.***>

**My working day may not be your working day. Please do not feel obliged to reply to this email outside of your normal working hours.**



From: Don Thornbury ***@***.***>
Reply-To: pulibrary/aspace_helpers ***@***.***>
Date: Thursday, August 4, 2022 at 2:27 PM
To: pulibrary/aspace_helpers ***@***.***>
Cc: "Regine I. Heberlein" ***@***.***>, Mention ***@***.***>
Subject: Re: [pulibrary/aspace_helpers] Generating MARC records from C0140 EAD component records/archival objects in Resource records (#181)


(After lurking a long time)
Regarding Lynn's comment about holdings: I believe there has to be "inventory" of some kind for a record to display in Blacklight. In this case the holdings will be kind of fake, but I think it still has to be there.

**Comment by faithc** [comment]
> I don't think we are looking to the relator values in this case, but I would like @faithc to confirm.

I think you're right, @LynnDurgin.

**Comment by regineheberlein** [comment]
@faithc @LynnDurgin 
Three leaf components don't currently have containers associated with them:

aspace_C0140_c74364-56855
aspace_C0140_c72729-43518
aspace_C0140_c87678-60535

Is that a data issue? They're currently being skipped by the transformation.

**Comment by faithc** [comment]
@regineheberlein 

1. aspace_C0140_c74364-56855 is a digital file so wouldn't have a container. 
2. There was a data issue with aspace_C0140_c72729-43518; I've corrected it.
3. aspace_C0140_c87678-60535 does have a container associated with it (Box P-000190). 

Can the transformation be tweaked to account for components without containers?

**Comment by regineheberlein** [comment]
@faithc 
Can you double-check on aspace_C0140_c87678-60535? It exports without a container. Also, it seems to be set to "internal", so should it be skipped regardless?

```
<c audience="internal" id="aspace_C0140_c87678-60535" level="file">
				<did>
					<unittitle>Francis Ray Farm Journal</unittitle>
					<unitid type="aspace_uri">/repositories/5/archival_objects/1459050</unitid>
					<origination label="Creator"><persname role="cre" rules="rda" source="local"
							>Ray, Francis</persname></origination>
					<physdesc altrender="whole"><extent altrender="materialtype spaceoccupied">1
							box</extent></physdesc>
					<unitdate calendar="gregorian" datechar="creation" era="ce" normal="1858/1890"
						type="inclusive">1858-1890</unitdate>
				</did>
				<acqinfo id="aspace_d20d96df5e24376be262e76e443a005f">
					<head>Immediate Source of Acquisition</head>
					<p>Purchased from Dan Casavant Rare Books in 2022 (AM-2022-112).</p>
				</acqinfo>
				<processinfo id="aspace_7086e38bad5689a445dc4d68456fb4f8">
					<head>Processing Information</head>
					<p>These materials were processed and described by Phoebe Nobles in March, 2022,
						incorporating some description provided by the dealer.</p>
				</processinfo>
				<scopecontent id="aspace_92b5d4f3f2c99a0b50568c22be65b58e">
					<head>Scope and Contents</head>
					<p>The partial journal kept by Francis Ray and his son Josiah T. Ray. It
						includes accounts of shoe sales (possibly shoes for plow animals), accounts
						of farm produce such as corn, tomatoes and onions, and some sales of honey,
						whiskey, brandy and guns, as well as accounts for day labor. The journal
						contains a last will and testament for Francis Ray, leaving his estate to
						Josiah Ray and requesting that Josiah support his mother Elizabeth Ray; the
						will also mentions daughters Catherine Ray and Nellie Ray. The journal comes
						from the Hawfields, a region in Thompson Township, Alamance County, North
						Carolina, where both people with West African ancestry and Ulster Irish
						immigrants settled. The journal likely includes records of Black customers
						and possibly those who were formerly enslaved (possibly including Charles
						Minor, Olson Stanford, and Frank Morrow). Oak Grove Plantation was
						nearby.</p>
				</scopecontent>
				<phystech audience="internal" id="aspace_d570f15c15e638da9e11a12245d46963">
					<head>Physical Characteristics and Technical Requirements</head>
					<p>In conservation receiving treatment; fragile condition.</p>
				</phystech>
				<controlaccess>
					<subject source="lcsh">Agricultural laborers -- Southern States -- History --
						19th century.</subject>
					<subject source="lcsh">African American agricultural laborers -- Southern States
						-- History -- 19th century.</subject>
					<geogname source="lcnaf">Alamance County (N.C.)</geogname>
					<genreform source="aat">Ledgers (account books) -- 19th century</genreform>
				</controlaccess>
			</c>
```

**Comment by regineheberlein** [comment]
@faithc 

C0140_c74364-56855 has no instances--just double-checking that's as it should be, since you're calling it a digital file.

**Comment by faithc** [comment]
> 

Sooo, this is bizarre. The archival object record looks totally fine to me as does the top container record. And I don't even know how a record can be set to "internal." Everything is marked "Publish."

**Comment by faithc** [comment]
> @faithc Can you double-check on aspace_C0140_c87678-60535? It exports without a container. Also, it seems to be set to "internal", so should it be skipped regardless?
> 
> ```
> <c audience="internal" id="aspace_C0140_c87678-60535" level="file">
> 				<did>
> 					<unittitle>Francis Ray Farm Journal</unittitle>
> 					<unitid type="aspace_uri">/repositories/5/archival_objects/1459050</unitid>
> 					<origination label="Creator"><persname role="cre" rules="rda" source="local"
> 							>Ray, Francis</persname></origination>
> 					<physdesc altrender="whole"><extent altrender="materialtype spaceoccupied">1
> 							box</extent></physdesc>
> 					<unitdate calendar="gregorian" datechar="creation" era="ce" normal="1858/1890"
> 						type="inclusive">1858-1890</unitdate>
> 				</did>
> 				<acqinfo id="aspace_d20d96df5e24376be262e76e443a005f">
> 					<head>Immediate Source of Acquisition</head>
> 					<p>Purchased from Dan Casavant Rare Books in 2022 (AM-2022-112).</p>
> 				</acqinfo>
> 				<processinfo id="aspace_7086e38bad5689a445dc4d68456fb4f8">
> 					<head>Processing Information</head>
> 					<p>These materials were processed and described by Phoebe Nobles in March, 2022,
> 						incorporating some description provided by the dealer.</p>
> 				</processinfo>
> 				<scopecontent id="aspace_92b5d4f3f2c99a0b50568c22be65b58e">
> 					<head>Scope and Contents</head>
> 					<p>The partial journal kept by Francis Ray and his son Josiah T. Ray. It
> 						includes accounts of shoe sales (possibly shoes for plow animals), accounts
> 						of farm produce such as corn, tomatoes and onions, and some sales of honey,
> 						whiskey, brandy and guns, as well as accounts for day labor. The journal
> 						contains a last will and testament for Francis Ray, leaving his estate to
> 						Josiah Ray and requesting that Josiah support his mother Elizabeth Ray; the
> 						will also mentions daughters Catherine Ray and Nellie Ray. The journal comes
> 						from the Hawfields, a region in Thompson Township, Alamance County, North
> 						Carolina, where both people with West African ancestry and Ulster Irish
> 						immigrants settled. The journal likely includes records of Black customers
> 						and possibly those who were formerly enslaved (possibly including Charles
> 						Minor, Olson Stanford, and Frank Morrow). Oak Grove Plantation was
> 						nearby.</p>
> 				</scopecontent>
> 				<phystech audience="internal" id="aspace_d570f15c15e638da9e11a12245d46963">
> 					<head>Physical Characteristics and Technical Requirements</head>
> 					<p>In conservation receiving treatment; fragile condition.</p>
> 				</phystech>
> 				<controlaccess>
> 					<subject source="lcsh">Agricultural laborers -- Southern States -- History --
> 						19th century.</subject>
> 					<subject source="lcsh">African American agricultural laborers -- Southern States
> 						-- History -- 19th century.</subject>
> 					<geogname source="lcnaf">Alamance County (N.C.)</geogname>
> 					<genreform source="aat">Ledgers (account books) -- 19th century</genreform>
> 				</controlaccess>
> 			</c>
> ```

Sooo, this is bizarre. The archival object record looks totally fine to me as does the top container record. And I don't even know how a record can be set to "internal." Everything is marked "Publish."

**Comment by regineheberlein** [comment]
How about C0140_c74364-56855? Is that all right?

**Comment by faithc** [comment]
Yeah, I confirmed with Kelly that it's a digital object that's not online, so it's fine. Thanks for checking in about it.

**Comment by regineheberlein** [comment]
@faithc @LynnDurgin 
Ok, we're waiting for the Alma Sandbox refresh to import the records, but meanwhile they're ready for you to poke at:
[C0140_out.txt](https://github.com/pulibrary/aspace_helpers/files/9330067/C0140_out.txt)

**Comment by regineheberlein** [comment]
@faithc @LynnDurgin 
The records are now available in the Alma Sandbox

**Comment by regineheberlein** [comment]
@faithc @LynnDurgin 
Have you had a chance to look at these so we can import them to production and close the ticket?

**Comment by faithc** [comment]
@regineheberlein Sorry- I was out for a while. Doing some QC now!

**Comment by regineheberlein** [comment]
Welcome back @faithc ! Are these ready to go into production?

**Comment by faithc** [comment]
Thanks, @regineheberlein. After doing some QC, the Firestone team noted a few issues/questions: 

- Is there a way to see what the records would look like in the catalog?
- Items are identified as a Book rather than a Manuscript
- Will digital objects/content display in the catalog record? Examples: C0140_c97187-07864, C0140_c01027-94291 
- This one seems to have something weird going on with the creator getting put into a 700 field (with a lot of additional data in it) rather than 100: C0140_c85700-18433, also  
- This one also has the creator in 700 instead of 100 but doesn’t have the extra data: C0140_c80184-00264

Thanks!

**Comment by regineheberlein** [comment]
Thanks!

re: #1: Yes, we can index a few records in catalog staging.

re: #2: Do you have an example? They're hardcoded as "mixed materials", so that's surprising. There are some pre-existing records in the sandbox that you may have been looking at. I've deleted all but two from the sandbox (those two are linked so I don't want to touch them) and will re-import so you can confirm.

re: #3: at this time, we're not including ASpace instances. we can always add them later; feel free to put in a ticket.

re: #4 and #5: yikes, that's a bug that's affecting 3 records. I'll take a look.

**Comment by faithc** [comment]
re: #1: Thanks, Regine. 

re: #2: Two examples: C0140_c00001 and C0140_c76598-42862

re: #3: Gotcha. 

re: #4: Thanks! 

FYI, I'll be out for the rest of the week starting tomorrow so I won't be able to return to this until next week.

**Comment by regineheberlein** [comment]
Thanks for the heads up. Who should I tag in your absence?

**Comment by faithc** [comment]
You can tag @kellybolding or @LynnDurgin should any questions arise while I'm away, but this may need to wait until I get back. Thanks.

**Comment by regineheberlein** [comment]
@kellybolding @LynnDurgin @faithc 

The records are updated in the Alma Sandbox. You can see them by going to Admin --> Manage Sets --> C0140_QA [don't open the link--select "Results" from the `...` menu]

**Comment by regineheberlein** [comment]
Thanks! I was just looking at the agents myself thanks to a note from DT, that's definitely a bug.

**Comment by regineheberlein** [comment]
Interesting catch about the amersand! I was ready to suspect Alma, but it happens on the export end. I'll see what if anything can be done about it.

**Comment by regineheberlein** [comment]
Both the agents and the ampersand should be fixed, as are dates in 008 / 046 and material type in leader (thanks, DT!). Take another look?

**Comment by regineheberlein** [comment]
@kellybolding @LynnDurgin You can preview the records on catalog-staging.princeton.edu now. The best way to get at the relevant records is to filter by "recently added".

**Comment by regineheberlein** [comment]
Thanks @kellybolding @LynnDurgin @doncat for the QA! @faithc waiting for your approval to move these to production.

**Comment by faithc** [comment]
Hi @regineheberlein - Thanks for your work on this. I was able to review some records and noticed just a couple of issues:

1. For format, the records note: "Manuscript, Book." Examples: C0140_c00001 and C0140_c03358
2. This is fairly minor, but is there a way to add a comma between the title and the date of creation? It's confusing when creator life dates are included in the title in particular. Examples: C0140_c00001 and C0140_c03358.

Of more concern is the missing processinfo note, though Kelly mentioned that you created a ticket to index these. Is that correct?

**Comment by regineheberlein** [comment]
Thanks @faithc ! 
The format display is correct, even though it is confusing. I had to ask DT about it myself. It's the display for resources coded as mixed materials.

I'm afraid adding a comma after the title is out of scope for this project--that would be a ticket in orangelight, which handles the display. 

You are correct: the processinfo note is being investigated on a separate ticket. https://github.com/pulibrary/bibdata/issues/1977

**Comment by faithc** [comment]
Ah ok, gotcha re: the format display and title punctuation. 

And great; thanks re: processinfo.

From my quick chat with Kelly, it seems like we should also create a ticket so that viaf names and non-LC SH, like those from the Homosaurus, are indexed and discoverable?

**Comment by regineheberlein** [comment]
Yup, you could open one in bibdata if you want to request those names to be indexed: https://github.com/pulibrary/bibdata

That would be a new feature, so I'll leave it to you to request it.

**Comment by faithc** [comment]
Request to index non-LC names in records, i.e. those designated as "viaf" or "uncontrolled."

**Comment by faithc** [comment]
Request to change 6xx $2 to $0 so that names associated with "viaf," uncontrolled," or "local sources" are indexed.

**Comment by regineheberlein** [comment]
As discussed off-line today, we'll explore doing this where a $0 is present

**Comment by regineheberlein** [comment]
Also decided in the same conversation toay, for "local" headings, we'll change ind1 to 4 and delete $2

**Comment by regineheberlein** [comment]
@faithc 
Are these items blockers for loading the records to production in the meantime?

**Comment by faithc** [comment]
They are not. Thanks.

**Comment by regineheberlein** [comment]
This is done, please review on catalog-staging

**Comment by regineheberlein** [comment]
blocked by ongoing discussion regarding 6xx handling

**Comment by faithc** [comment]
looping in @phoebeno

**Comment by faithc** [comment]
@regineheberlein Please revert the most recent work you did so that the 600 and 610 fields retain the ind2 = 7s with $2 and load these records into production. 

There will be a forthcoming proposal to DSSG to add 6xx ind2=7 with $2 to the index so that these fields display.

**Comment by regineheberlein** [comment]
@faithc The records are available for another round of QA on catalog-staging.

**Comment by faithc** [comment]
Aside from the processing note not appearing, which you previously noted is being addressed in a separate ticket, these look good to me. Thanks, @regineheberlein.

**Comment by regineheberlein** [comment]
Great! I'll import them tonight then.

**Comment by regineheberlein** [comment]
This is done
